pkg/kcp/provider/gcp/meta/metadata.go

```diff
@@ -74,3 +75,21 @@ func IsTooManyRequests(err error) bool {
 
 	return false
 }
+
+func IsOperationInProgressError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var googleapierror *googleapi.Error
+	if ok := errors.As(err, &googleapierror); ok {
+		if googleapierror.Code == 400 {
+			for _, e := range googleapierror.Errors {
+				if strings.Contains(e.Message, "peering operation in progress") {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
```

Comment:
Since this func is in a shared metadata file, `IsOperationInProgressError` name is misleading.
It is specifically tied to peering operation in progress.